### PR TITLE
Setting ContentType in the HTTP responses

### PR DESCRIPTION
**Pull request by moozzyk**
This fixes the issue #1106 where Firefox tries to parse all responses without Content-Type as Xml and displays the 'XML Parsing Error: no root element found' warning in the dev console.

Also making auth test work in IE

### CODE REVIEW

**Comment by BrennanConroy** [comment]
I believe this also happens for 'No connection' errors

File: src/Microsoft.AspNetCore.Sockets.Http/HttpConnectionDispatcher.cs [hunk 1]
```diff
@@ -433,6 +433,8 @@ private static string GetNegotiatePayload(string connectionId, HttpSocketOptions
                     return;
                 }
             }
+
+            context.Response.ContentType = "text/plain";
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Sockets.Http/HttpConnectionDispatcher.cs]: Hmm

**Comment by BrennanConroy** [comment]
🆙 📅

**Comment by BrennanConroy** [comment]
Ping @anurse @davidfowl @mikaelm12

**Comment by anurse** [comment]
Looks OK as long as it's still relevant and rebased properly.

**Comment by BrennanConroy** [comment]
Thanks for your contribution @moozzyk ! ;)

**Comment by moozzyk** [comment]
@BrennanConroy :trollface:

**Comment by muratg** [comment]
🤣

**Pull request merged**